sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -796,3 +797,74 @@ case class SchemaOfJson(
 
   override def prettyName: String = "schema_of_json"
 }
+
+/**
+ * A function which returns all the keys of the outmost JSON object.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(json_object) - returns all the keys of the outmost JSON object as an array.",
+  arguments = """
+    Arguments:
+      * json_object - A JSON object. If a valid JSON object is given, all the keys of the outmost
+          object will be returned as an array. If it is an invalid JSON string, the function
+          returns null. If it is a JSON array or empty string, an illegal argument exception
+          will be thrown.
```

Comment:
I'm confused. Why the behavior is different between invalid JSON and JSON array/empty string?

Reply:
For invalid JSON we are returning `NULL` in all the other json functions. So for that i kept it like that.

Reply:
Then we should return null for JSON array or empty string, as they are also invalid as the input of this function.

Reply:
Ok, i will make the required changes here.
In that case, i guess i need to do the same in #27759 too ?

Reply:
yea